### PR TITLE
Instrument tests independent of platform

### PR DESCRIPTION
**Pull request by stavros11**
Currently instrument tests are using custom test runcards with hardcoded IPs, so they will not work in any slurm queue. This PR removes these runcards and uses the platform runcards which are in agreement with the queue of the same name. If some instrument is not available in the used queue, the corresponding tests will be skipped.

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

File: src/qibolab/tests/test_instruments_qblox.py [hunk 1]
```diff
@@ -108,17 +98,18 @@ def test_instruments_qublox_set_property_wrappers(name):
...
-            "discretization_threshold_acq",
-            sequencer,
-            "discretization_threshold_acq",
-            values=np.linspace(-16777212.0, 16777212.0, 20),
-        )
-        instrument_test_property_wrapper(
-            instrument, "phase_rotation_acq", sequencer, "phase_rotation_acq", values=np.linspace(0, 359, 20)
-        )
-    if name == "ClusterQCM_RF":
+        # FIXME: I don't know why this is failing
```
Comment (scarrazza) [inline src/qibolab/tests/test_instruments_qblox.py:101]: I see coverage going down and these comments here, are you planning to look at these before merging?
I believe the original PR has a new commit (https://github.com/qiboteam/qibolab/pull/255/commits/c43d9209078dbb7cc93cad1fef7d91be05ae4b0d) which might have some impact.

Comment (scarrazza) [reply src/qibolab/tests/test_instruments_qblox.py:101]: Thanks, please retry now, the IP has changed.

File: src/qibolab/tests/conftest.py [hunk 2]
```diff
@@ -4,24 +4,20 @@
...
-
-    if metafunc.module.__name__ == "qibolab.tests.test_instruments_rohde_schwarz":
-        pytest.skip("Skipping Rohde Schwarz tests because it is not available in qpu5q.")
+    platform_name = metafunc.config.option.platform
 
     if "platform_name" in metafunc.fixturenames:
         if "qubit" in metafunc.fixturenames:
             # TODO: Do backend initialization here instead of every test (currently does not work)
-            qubits = [(n, q) for n in platforms for q in range(Platform(n).nqubits)]
+            qubits = [(platform_name, q) for q in range(Platform(platform_name).nqubits)]
```
Comment (aorgazf) [inline src/qibolab/tests/conftest.py:20]: Hi @stavros11,
to support qubits with string IDs, I would use:
```python
qubits = [(platform_name, q) for q in Platform(platform_name).qubits]
```

Comment (stavros11) [reply src/qibolab/tests/conftest.py]: Done, thanks.

**Comment by stavros11** [comment]
@scarrazza this implements a fix for a bug that different people have reported to me for some time (two_qubit_gate argument missing from the transpiler) as well as a few more backend fixes after refactor 1. It also allows to execute tests on tii1q_b1 using the labels. Please have a look as I think it should be merged.

**Review by scarrazza — APPROVED** [review]

**Comment by scarrazza** [comment]
Looks good to me, could you please fix tests so we can merge?

**Comment by stavros11** [comment]
> Looks good to me, could you please fix tests so we can merge?

Thanks, I fixed the tests, however there are some issues introduced by the new workflows. In particular the [options added in pyproject.toml](https://github.com/qiboteam/qibolab/blob/f7e1bba97cafb398773ed179b34c915acd68b1cf/pyproject.toml#L8) are not valid for the tests executed by `selfhosted.yml`, which is what this PR is fixing. There we need to use `-m qpu` (instead of `-m "not qpu"`) and the platform should not be hardcoded to `qili1q_os2` but should be read from the label.

**Comment by scarrazza** [comment]
In principle this should work now, right?

**Comment by stavros11** [comment]
> In principle this should work now, right?

Yes, it should not be blocked anymore, I just need to fix the conflicts. 

However, given that now for the first time we have two working platforms on main that are not both qblox, I will give a try to enable tests for both and see if they are really platform independent. We can also check the labels (run-on-qili1q_os2, run-on-qw5q_gold). So I would postpone merging it until this is done.

**Comment by scarrazza** [comment]
Perfect, thanks.

**Comment by stavros11** [comment]
Tests passing on both qili1q_os2 and qw5q_gold, including the selfhosted workflow on both qpus.

Note that conftest.py and some of the instrument tests are a bit messy, but this is mostly because currently we are trying to accommodate both the new `DesignPlatform` and the `MultiqubitPlatform`. When we refactor everything to use the `InstrumentDesign`, testing should also be simpler. For now I think this is ready.

**Review by scarrazza — APPROVED** [review]
Great, we can merge.

**Pull request merged**